uniffi_bindgen/src/bindings/kotlin/gen_kotlin/miscellany.rs

```diff
@@ -58,15 +58,15 @@ macro_rules! impl_code_type_for_miscellany {
 
 impl_code_type_for_miscellany!(
     TimestampCodeType,
-    "java.time.Instant",
+    "Instant",
     "Timestamp",
     vec!["java.time.Instant", "java.time.DateTimeException"],
     "TimestampHelper.kt"
 );
 
 impl_code_type_for_miscellany!(
     DurationCodeType,
-    "java.time.Duration",
+    "Duration",
```

Comment:
Couldn't comment where the actual fix needed so commenting here. 

This changes the `$type_label`, instead of using the `$canonical_name`.

That should be used in the `canonical_name` method, but that's missing. 

The correct fix is to add that method (say at [L24][1]).

```rust
                 fn canonical_name(&self, _oracle: &dyn CodeOracle) -> String {
                     $canonical_name.into()
                 }
```

The fully qualified class name should be available as a type label, so as to avoid collisions with UDL declared types. e.g. if a UDL type is called `Instant` or `Duration`.

[1]: https://github.com/mozilla/uniffi-rs/pull/1114/files#diff-326998556286e54771636144110306a7bf185b8d277ff93208998226cb90116cR24

Reply:
Adding the canonical_name method makes a lot of sense.  If the type name is going to be fully qualified, then should we remove the imports?

Reply:
I don't quite understand the concrete suggestion here, and if the intent is to allow a `Duration` type to be defined in a UDL and not conflict with this `Duration`, I think things will need to get even more sophisticated - `writeOptionalInstant` will conflict - so this case will want to do something like `writeOptionaljava_time_Instant`.

Given these problems already exist, that this is fixing a regression which is preventing app-services from upgrading, and because this change doesn't prevent fixing this correctly, I've decided to merge this as is, and I'll open an issue for that "correct" fix.